src/devices_models/devices/common/duration_constraints.jl

```diff
@@ -196,11 +196,12 @@ function device_duration_look_ahead!(
             lhs_off = JuMP.GenericAffExpr{Float64, JuMP.VariableRef}(0)
             for i in UnitRange{Int}(Int(t - duration_data[ix].down + 1), t)
                 if i in time_steps
-                    JuMP.add_to_expression!(lhs_off, (1 - varon[name, i]))
+                    JuMP.add_to_expression!(lhs_off, 1)
+                    JuMP.add_to_expression!(lhs_off, -1, varon[name, i])
```

Comment:
the previous allocates more because `(1 - varon[name, i])` becomes a new expression